pkg/controller/datavolume-controller.go

```diff
@@ -1222,22 +1371,78 @@ func (r *DatavolumeReconciler) snapshotSmartClonePossible(dataVolume *cdiv1.Data
 	return true, nil
 }
 
-func (r *DatavolumeReconciler) getCloneStrategy() (cdiv1.CDICloneStrategy, error) {
+func (r *DatavolumeReconciler) getCloneStrategy(dataVolume *cdiv1.DataVolume) (*cdiv1.CDICloneStrategy, error) {
+	defaultCloneStrategy := cdiv1.CDICloneStrategy(cdiv1.CloneStrategySnapshot)
+	sourcePvc, err := r.findSourcePvc(dataVolume)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.recorder.Eventf(dataVolume, corev1.EventTypeWarning, ErrUnableToClone, "Source pvc %s not found", dataVolume.Spec.Source.PVC.Name)
+		}
+		return nil, err
+	}
+	storageClass, err := GetStorageClassByName(r.client, sourcePvc.Spec.StorageClassName)
+	if err != nil {
+		return nil, err
+	}
+
+	strategyOverride, err := getGlobalCloneStrategyOverride(r)
+	if err != nil {
+		return nil, err
+	}
+	if strategyOverride != nil {
+		return strategyOverride, nil
+	}
+
+	// do check storageProfile and apply the preferences
+	strategy, err := getPreferredCloneStrategyForStorageClass(r.client, storageClass)
+	if err != nil {
+		return nil, err
+	}
+	if strategy != nil {
+		return strategy, err
+	}
+
+	return &defaultCloneStrategy, nil
+}
+
+func (r *DatavolumeReconciler) findSourcePvc(dataVolume *cdiv1.DataVolume) (*corev1.PersistentVolumeClaim, error) {
+	sourcePvcSpec := dataVolume.Spec.Source.PVC
+	if sourcePvcSpec == nil {
+		return nil, errors.New("no source PVC provided")
+	}
+
+	// Find source PVC
+	sourcePvcNs := sourcePvcSpec.Namespace
+	if sourcePvcNs == "" {
+		sourcePvcNs = dataVolume.Namespace
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: sourcePvcNs, Name: sourcePvcSpec.Name}, pvc); err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.log.V(3).Info("Source PVC is missing", "source namespace", sourcePvcSpec.Namespace, "source name", sourcePvcSpec.Name)
+		}
+		return nil, err
+	}
+	return pvc, nil
+}
+
+func getGlobalCloneStrategyOverride(r *DatavolumeReconciler) (*cdiv1.CDICloneStrategy, error) {
```

Comment:
If the only value you are passing into this function is the reconciler, why not make it:
```go
func (r *DatavolumeReconciler) getGlobalCloneStrategyOverride() (*cdiv1.CDICloneStrategy, error) {
...
}
```
Seems more logical to me.

Reply:
Done